bftengine/src/bftengine/KeyExchangeManager.cpp

```diff
@@ -131,6 +133,37 @@ void KeyExchangeManager::sendKeyExchange(const SeqNum& sn) {
   metrics_->sent_key_exchange_counter.Get().Inc();
 }
 
+// sends the clients public keys via the internal client, if keys weren't published or outdated.
+void KeyExchangeManager::sendInitialClientsKeys(const std::string& keys) {
+  if (clientsPublicKeys_.published_) {
+    LOG_INFO(KEY_EX_LOG, "Clients public keys were already published");
+    return;
+  }
+  auto ret = std::async(std::launch::async, [this, keys]() {
+    while (!initial_exchange_) {
```

Comment:
why not to use an atomic bool and not exchanged() only?

Reply:
exchanged is not thread-safe no?